mautic/mautic
# FIX: Fixes form theme used for rendering

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Partly contributes to #13034  



#### Description:

FormModel::generateHtml now uses the system-wide default theme from the core parameters (`Configuration > Theme Settings > Default theme`) if no specific template is provided.

This change ensures to fallback to the system-wide default theme, before using `@MauticForm/Builder` as last resort.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Set `Configuration > Theme Settings > Default theme` to `Neopolitan`
3. Create a form; do **not** set a specific template
4. View the standalone form https://mautic.xyz/form/ID

Before this PR Mautic will use the fallback style file [FormBundle/Resources/views/Builder/_style.html.twig](https://github.com/mautic/mautic/blob/7a39f3954909e52dd6b36ba478d29d477e06b9c1/app/bundles/FormBundle/Resources/views/Builder/_style.html.twig).

After this PR will use Neopolitans style.html.twig. If the default theme doesn't come with this file, Mautic will use [FormBundle/Resources/views/Builder/_style.html.twig](https://github.com/mautic/mautic/blob/7a39f3954909e52dd6b36ba478d29d477e06b9c1/app/bundles/FormBundle/Resources/views/Builder/_style.html.twig) as last fallback.
